deeppavlov/models/ranking/siamese_model.py

```diff
@@ -0,0 +1,112 @@
+# Copyright 2017 Neural Networks and Deep Learning lab, MIPT
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np
+from typing import List, Iterable, Union
+
+from deeppavlov.core.models.nn_model import NNModel
+
+
+class SiameseModel(NNModel):
+    """The class implementing base functionality for siamese neural networks.
+
+    Args:
+        batch_size: A size of a batch.
+        num_context_turns: A number of ``context`` turns in data samples.
+        *args: Other parameters.
+        **kwargs: Other parameters.
+    """
+
+    def __init__(self,
+                 batch_size: int,
+                 num_context_turns: int = 1,
+                 *args,
+                 **kwargs):
+
+        self.save_path = kwargs.get('save_path', None)
+        self.load_path = kwargs.get('load_path', None)
+        mode = kwargs.get('mode', None)
+
+        super(SiameseModel, self).__init__(save_path=self.save_path, load_path=self.load_path, mode=mode)
+
+        self.batch_size = batch_size
+        self.num_context_turns = num_context_turns
+
+    def load(self, *args, **kwargs) -> None:
+        pass
+
+    def save(self, *args, **kwargs) -> None:
+        pass
+
+    def train_on_batch(self, batch: List[List[np.ndarray]], y: List[int]) -> float:
+        b = self._make_batch(batch)
+        loss = self._train_on_batch(b, y)
+        return loss
+
+    def __call__(self, batch: Iterable[List[np.ndarray]]) -> Union[np.ndarray, List[str]]:
+        y_pred = []
+        buf = []
+        j = 0
+        while True:
+            try:
+                el = next(batch)
+                j += 1
```

Comment:
`for j, el in enumerate(batch, start=1):`?

Reply:
corrected